dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AdminPrivilegesFilter.java

```diff
@@ -1,34 +1,56 @@
 package io.dockstore.webservice.resources;
 
+import java.lang.reflect.Method;
 import java.security.Principal;
 
+import javax.annotation.security.RolesAllowed;
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
 import javax.ws.rs.ext.Provider;
 
+import io.dockstore.webservice.SimpleAuthorizer;
 import io.dockstore.webservice.core.User;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Every request will be filtered through here to determine if an admin user is making the request. If it is an admin,
- * then the username, method, and endpoint path will be logged.
+ * and it is a request that requires admin privileges, then the username, method, and endpoint path will be logged.
  */
 @Provider
 public class AdminPrivilegesFilter implements ContainerRequestFilter {
     private static final Logger LOG = LoggerFactory.getLogger(AdminPrivilegesFilter.class);
 
+    @Context
+    private ResourceInfo resourceInfo;
+
     @Override
     public void filter(ContainerRequestContext requestContext) {
         if (requestContext != null) {
-            Principal principal = requestContext.getSecurityContext().getUserPrincipal();
+            final Principal principal = requestContext.getSecurityContext().getUserPrincipal();
             if (principal instanceof User) {
-                User user = (User)principal;
-                if (user.getIsAdmin()) {
-                    String logMessage = "Admin " + user.getUsername() + " making " + requestContext.getMethod() + " request at " + requestContext.getUriInfo().getPath();
+                final User user = (User)principal;
+                if (user.getIsAdmin() && requestRequiresAdminRole()) {
+                    final String logMessage = "Admin " + user.getUsername() + " making " + requestContext.getMethod()
```

Comment:
I think we should log the user id as well now. In case one of us changes our username.